File: app/src/main/java/com/doubean/ford/data/vo/Post.java
package com.doubean.ford.data.vo;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.SerializedName;

import java.time.LocalDateTime;
import java.util.List;

/**
 * Class officially known as topic/话题
 */
@Entity
public class Post {
    @NonNull
    @PrimaryKey
    @SerializedName("id")
    public String id;

    public String groupId;

    public String tagId;

    public String title;

    public User author;

    @SerializedName("create_time")
    public LocalDateTime created;

    @SerializedName("update_time")
    public LocalDateTime lastUpdated;

    @SerializedName("like_count")
    public Integer likeCount;

    @SerializedName("reactions_count")
    public Integer reactionCount;

    @SerializedName("reshares_count")
    public Integer repostCount;

    @SerializedName("collections_count")
    public Integer saveCount;

    @SerializedName("comments_count")
    public Integer commentCount;

    @SerializedName("abstract")
    public String shortContent;

    public String content;

    @SerializedName("topic_tags")
    public List<GroupPostTag> postTags;

    @SerializedName("cover_url")
    public String coverUrl;

    public String url;

    public GroupBrief group;

    public String uri;

    public String getTagName() {
        if (postTags.isEmpty()) return null;
        return postTags.get(0).name;
    }
}

File: app/src/main/java/com/doubean/ford/data/vo/Post.kt
package com.doubean.ford.data.vo

import androidx.room.*
import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

/**
 * Class officially known as topic/话题
 */
@Entity(tableName = "posts")
data class Post (
    @PrimaryKey
    @SerializedName("id")
    val id: String,

    val title: String,

    val author: User,


    @SerializedName("create_time")
    val created: LocalDateTime? = null,

    @SerializedName("update_time")
    val lastUpdated: LocalDateTime? = null,


    @SerializedName("like_count")
    val likeCount: Int? = null,

    @SerializedName("reactions_count")
    val reactionCount: Int? = null,


    @SerializedName("reshares_count")
    val repostCount: Int? = null,


    @SerializedName("collections_count")
    val saveCount: Int? = null,


    @SerializedName("comments_count")
    val commentCount: Int? = null,

    @SerializedName("abstract")
    val shortContent: String? = null,

    val content: String?,


    @SerializedName("topic_tags")
    val postTags: List<GroupPostTag>? = null,

    @SerializedName("cover_url")
    val coverUrl: String? = null,

    val url: String,

    val group: GroupBrief?=null,

    val uri: String,

    ){
    var groupId: String? = null
    var tagId: String?=null
    val tagName: String?
        get() = if (postTags!!.isEmpty()) null else postTags[0].name
}